modules/apps/object/object-service/src/main/java/com/liferay/object/internal/action/engine/ObjectActionEngineImpl.java

```diff
@@ -73,6 +74,9 @@ public void executeObjectAction(
 
 		_executeObjectAction(
 			objectAction, objectDefinition, payloadJSONObject, userId,
+			ObjectEntryVariablesUtil.getActionOldVariables(
+				_dtoConverterRegistry, objectDefinition, payloadJSONObject,
+				_systemObjectDefinitionMetadataRegistry),
 			ObjectEntryVariablesUtil.getActionVariables(
 				_dtoConverterRegistry, objectDefinition, payloadJSONObject,
 				_systemObjectDefinitionMetadataRegistry));
```

Comment:
@p-albuquerque can you try to change the logic of `getActionVariables` to return an unique map like:

```
HashMapBuilder.put(
	"currentVariables", ....
).put(
	"oldVariables", ....
).build();
```
?

This way you don't need to call two methods every time and you can do `map.get("oldVariables")` or `map.get("currentVariables")` when necessary